File: org.opencompare/io-wikipedia/src/main/scala/org/opencompare/io/wikipedia/pcm/Cell.scala
package org.opencompare.io.wikipedia.pcm

import org.opencompare.api.java.impl.PCMFactoryImpl


class Cell(
  val content : String,
  val rawContent : String,
  val isHeader : Boolean,
  val row : Int,
  val rowspan : Int,
  val column : Int,
  val colspan : Int
  ) {
  
  override def toString() : String = {
     content
  }

  def toPCM() : org.opencompare.api.java.Cell = {
    val factory = new PCMFactoryImpl()
    val cell = factory.createCell()
    cell.setContent(content)
    cell
  }
  
}

File: org.opencompare/api-java/src/main/java/org/opencompare/api/java/io/Cell.java
package org.opencompare.api.java.io;

/**
 * Created by smangin on 02/07/15.
 */
public class Cell {

    private String content;
    private String rawContent;
    private boolean isHeader;
    private int row;
    private int column;
    private int rowspan;
    private int colspan;

    public Cell(String content, String rawContent, boolean isHeader, int row, int column, int rowspan, int colspan) {
        setContent(content);
        setRawContent(rawContent);
        setHeader(isHeader);
        setRow(row);
        setColumn(column);
        setRowspan(rowspan);
        setColspan(colspan);
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getRawContent() {
        return rawContent;
    }

    public void setRawContent(String rawContent) {
        this.rawContent = rawContent;
    }

    public boolean isHeader() {
        return isHeader;
    }

    public void setHeader(boolean isHeader) {
        this.isHeader = isHeader;
    }

    public int getRow() {
        return row;
    }

    public void setRow(int row) {
        this.row = row;
    }

    public int getColumn() {
        return column;
    }

    public void setColumn(int column) {
        this.column = column;
    }

    public int getRowspan() {
        return rowspan;
    }

    public void setRowspan(int rowspan) {
        this.rowspan = rowspan;
    }

    public int getColspan() {
        return colspan;
    }

    public void setColspan(int colspan) {
        this.colspan = colspan;
    }

    @Override
    public String toString() {
        return content;
    }
}
